### PR TITLE
Fix sample results transfer for two subapps

### LINKED ISSUE #17079: StochasticToolsTransfer does not transfer correctly when having two or more sub apps
## Bug Description

When two or more sub apps are provided, the StochasticToolsTransfer does not transfer correctly.

## Steps to Reproduce

Sampler multiapp looks for all transfers in warehouse, and it will try to transfer pp that even does not belong to it. The correct way is to only to find transfer that belongs to its multiapp. 

## Impact

No existing tests use two or more sub apps, so there is no impact.

### PR DESCRIPTION
**Pull request by jiangwen84**
closes #17079

### CODE REVIEW

File: modules/stochastic_tools/src/multiapps/SamplerFullSolveMultiApp.C [hunk 1]
```diff
@@ -186,7 +186,7 @@ SamplerFullSolveMultiApp::getActiveStochasticToolsTransfers(Transfer::DIRECTION
   for (std::shared_ptr<Transfer> transfer : warehouse.getActiveObjects())
   {
     auto ptr = std::dynamic_pointer_cast<StochasticToolsTransfer>(transfer);
-    if (ptr)
+    if (ptr && ptr->getMultiApp()->name() == this->name())
```
Comment (loganharbour) [inline modules/stochastic_tools/src/multiapps/SamplerFullSolveMultiApp.C:189]:
```suggestion
    if (ptr && ptr->getMultiApp().get() == this)
```

File: modules/stochastic_tools/src/multiapps/SamplerTransientMultiApp.C [hunk 2]
```diff
@@ -170,7 +170,7 @@ SamplerTransientMultiApp::getActiveStochasticToolsTransfers(Transfer::DIRECTION
   {
     std::shared_ptr<StochasticToolsTransfer> ptr =
         std::dynamic_pointer_cast<StochasticToolsTransfer>(transfer);
-    if (ptr)
+    if (ptr && ptr->getMultiApp()->name() == this->name())
```
Comment (loganharbour) [inline modules/stochastic_tools/src/multiapps/SamplerTransientMultiApp.C:173]:
```suggestion
    if (ptr && ptr->getMultiApp().get() == this)
```

File: modules/stochastic_tools/src/multiapps/SamplerFullSolveMultiApp.C (hunk 1 above)
Comment (loganharbour) [reply modules/stochastic_tools/src/multiapps/SamplerFullSolveMultiApp.C:189]: I'd say just compare the raw objects

Comment (jiangwen84) [reply modules/stochastic_tools/src/multiapps/SamplerFullSolveMultiApp.C]: Good point. Thanks.

**Comment by loganharbour** [comment]
Hah - I appreciate the faith in those suggestions compiling without testing it 😆

**Comment by jiangwen84** [comment]
@loganharbour I will be surprised if they would fail. Very nice suggestions :-)

**Review by loganharbour — APPROVED** [review]

**Pull request merged**